compiler/prelude/prelude.js

```diff
@@ -433,37 +433,52 @@ var $internalAppend = (slice, array, offset, length) => {
         return slice;
     }
 
-    var newArray = slice.$array;
-    var newOffset = slice.$offset;
     var newLength = slice.$length + length;
-    var newCapacity = slice.$capacity;
-
-    if (newLength > newCapacity) {
-        newOffset = 0;
-        newCapacity = Math.max(newLength, slice.$capacity < 1024 ? slice.$capacity * 2 : Math.floor(slice.$capacity * 5 / 4));
-
-        if (slice.$array.constructor === Array) {
-            newArray = slice.$array.slice(slice.$offset, slice.$offset + slice.$length);
-            newArray.length = newCapacity;
-            var zero = slice.constructor.elem.zero;
-            for (var i = slice.$length; i < newCapacity; i++) {
-                newArray[i] = zero();
-            }
-        } else {
-            newArray = new slice.$array.constructor(newCapacity);
-            newArray.set(slice.$array.subarray(slice.$offset, slice.$offset + slice.$length));
-        }
-    }
-
-    $copyArray(newArray, array, newOffset + slice.$length, offset, length, slice.constructor.elem);
+    slice = $growSlice(slice, newLength);
+    
+    var newArray = slice.$array;
+    $copyArray(newArray, array, slice.$offset + slice.$length, offset, length, slice.constructor.elem);
 
     var newSlice = new slice.constructor(newArray);
-    newSlice.$offset = newOffset;
+    newSlice.$offset = slice.$offset;
     newSlice.$length = newLength;
-    newSlice.$capacity = newCapacity;
+    newSlice.$capacity = slice.$capacity;
     return newSlice;
```

Comment:
Considering $growSlice already returns a new slice it seems suboptimal to create it once again. I understand that it's necessary to avoid modifying the original slice object when the slice didn't need growing, but it feels kind of inelegant.

Reply:
Added a check to see if `$growSlice` would create a new slice so that only one new slice is made. I felt like that would work better than trying to return a tuple, doing an "old slice/ new slice" comparison to see if it changed, or duplicating code. I like this better even though it does the same check twice. Is this more what you were thinking for a better solution?

Reply:
Would things get simpler if `$growSlice()` always returned a new slice, even if it wasn't grown? Not a big deal either way, just trying to streamline the code a bit.

Reply:
Me thinking it through: When [`func (v Value) Grow(n int)`](https://pkg.go.dev/reflect@go1.20.14#Value.Grow) is called with an $n$ such that $len + n \lt cap$, it has no effect the slice since the slice already has enough capacity to handle $len +n$. In that case having `$growSlice()` always return a new slice creates an unneeded slice, however we check for that case in the Go part of Grow prior to ever calling `$growSlice()`, so always returning a new slice won't effect Grow but would make the append logic simpler.

Okay, I'll make that change.

Reply:
Changed

Reply:
Yep, that looks much cleaner. Thanks!